test/psych/test_string.rb

```diff
@@ -24,10 +24,19 @@ def initialize
     # "ambiguity" in the emitted document
     def test_y_is_quoted
       assert_match(/"y"/, Psych.dump("y"))
+      assert_match(/"Y"/, Psych.dump("Y"))
     end
 
     def test_n_is_quoted
       assert_match(/"n"/, Psych.dump("n"))
+      assert_match(/"N"/, Psych.dump("N"))
+    end
```

Comment:
@tenderlove The spec below encompasses these 2 special cases, so I'd be happy to drop these 2 if you're good with it.

Reply:
Yes, the test you have below seems good enough

Reply:
Great, I dropped the 2 special tests, but left the comment (updated for uppercase): https://github.com/ruby/psych/pull/641/commits/6750b3540215e115c24bbeaaaca069b1508db72c